test/lit.cfg

```diff
@@ -568,6 +568,44 @@ if config.benchmark_o != 'Benchmark_O':
 # Add substitutions for the run target triple, CPU, OS, and pointer size.
 config.substitutions.append(('%target-triple', config.variant_triple))
 
+if platform.system() == 'Darwin':
+    # iOS 12.2 does not support 32-bit targets, so we cannot run tests that
+    # want to deploy to an iOS that has Swift in the OS.
+    if not (run_os == 'ios' and run_ptrsize == '32'):
+        config.available_features.add('swift_stable_abi')
+
+    PRE_STABLE_VERSION = {
+        'macosx': '10.14.3',
+        'ios': '12.1',
+        'appletvos': '12.1',
+        'watchos': '5.1'
+    }
+    pre_stable_version = PRE_STABLE_VERSION.get(run_os, '')
+    config.pre_stable_abi_triple = '%s-%s-%s%s' % (run_cpu, run_vendor, run_os,
+                                                   pre_stable_version)
+    STABLE_VERSION = {
+        'macosx': '10.14.4',
+        'ios': '12.2',
+        'appletvos': '12.2',
+        'watchos': '5.2'
+    }
+    stable_version = STABLE_VERSION.get(run_os, '')
+    config.stable_abi_triple = '%s-%s-%s%s' % (run_cpu, run_vendor, run_os,
+                                               stable_version)
+else:
+    config.pre_stable_abi_triple = config.variant_triple
+    config.stable_abi_triple = config.variant_triple
```

Comment:
Should `config.stable_abi_triple` be `NO_STABLE_ABI` or something on other platforms?

Reply:
I wanted to leave the possibility of writing tests that use stable ABI on Apple platforms but don't care elsewhere.

Reply:
Hm, okay.